core/cpu.cpp

```diff
@@ -618,9 +595,352 @@ u8 CPU::exec(){
+            u8 call_value = (opCode-0xC7);
+            pc = call_value;
+            return 16;
+        }
+        case 0xC8: {
+            //RET Z
+            if(readZeroFlag()){
+                pc = popFromStack();
+                return 20;
+            }
+            else{
+                return 8;
+            }
+        }
+        case 0xC9: {
+            //RET
+            pc = popFromStack();
+            return 16;
+        }
+        case 0xCA: {
+            //JP Z, a16
+            //PC NEEDS TO BE INCREMENTED TWICE ON 16 BIT READ
+            u16 nn_nn = mmu.read16Bit(pc++);
+            pc++;
+
+            if(readZeroFlag()){
+                pc = nn_nn;
+                return 16;
+            }
+            else{
+                return 12;
+            }
+        }
         case 0xCB: {
+            //It's a prefix function
             return execCB();
         }
+        case 0xCC: {
+            //CALL Z, a16
+            //check if zero flag is set
+
+            //PC NEEDS TO BE INCREMENTED TWICE ON 16 BIT READ
+            u16 nn_nn = mmu.read16Bit(pc++);
+            pc++;
+
+            if(readZeroFlag()){
+                pushToStack(pc);
+
+                pc = nn_nn;
+                return 24;
+            }
+            else{
+                return 12;
+            }
+            return 4;
+        }
+        case 0xCD: {
+            //CALL a16
+            pushToStack(pc);
+            
+            //PC NEEDS TO BE INCREMENTED TWICE ON 16 BIT READ
+            u16 nn_nn = mmu.read16Bit(pc++);
+            pc++;
+            pc = nn_nn;
+            return 24; 
+        }
+        case 0xCE: {
+            //ADC A, d8
+            u8 n = mmu.read(pc++);
+            u8 a = getHighByte(af);
+            u8 result = op_adc(a, n);
+            setHighByte(&af, result);
+            return 4;
+        }
+        case 0xD0: {
+            //RET NC
+            if(readCarryFlag() & readSubtractFlag()){
+                pc = popFromStack();
+                return 20;
+            }
+            else{
+                return 8;
+            }
+        }
+        case 0xD2: {
+            //JP NC, a16
+            //PC NEEDS TO BE INCREMENTED TWICE ON 16 BIT READ
+            u16 nn_nn = mmu.read16Bit(pc++);
+            pc++;
+
+            if(readCarryFlag() & readSubtractFlag()){
+                pc = nn_nn;
+                return 16;
+            }
+            else{
+                return 12;
+            }
+        }
+        case 0xD4: {
+            //CALL NC, a16
+            //check if subtraction flag(n) & carry flags are set
+
+            //PC NEEDS TO BE INCREMENTED TWICE ON 16 BIT READ
+            u16 nn_nn = mmu.read16Bit(pc++);
+            pc++;
+
+            if(readCarryFlag() & readSubtractFlag()){
+                pushToStack(pc);
+
+                pc = nn_nn;
+                return 24;
+            }
+            else{
+                return 12;
+            }
+            return 4;
```

Comment:
Unreachable

Reply:
I commented out the default case, I am guessing that is the intended behavior?